### PR TITLE
test: add coverage for Model API endpoints

### LINKED ISSUE #1270: Tests: Add API tests for new Model endpoints
## Description
Adds test coverage for the newly introduced Model APIs.

## Endpoints to cover

| Method | Endpoint |
|--------|----------|
| GET | `/models/status` |
| GET | `/models/hardware` |
| POST | `/models/setup` |
| POST | `/models/download/{model_name}` |
| GET | `/models/download/{task_id}/progress` |
| DELETE | `/models/{model_name}` |

## Tasks
- [ ] Write happy path + edge case tests for each endpoint

### PR DESCRIPTION
**Pull request by akshajtiwari**
###  Addressed Issues:

Fixes #1270 

###  Screenshots/Recordings:

TODO: If applicable, add screenshots or recordings that demonstrate the interface **before** and **after** the changes.

N/A

###  Additional Notes:
Removed the Bruno API collection as per request by the maintainers.
This was discussed on the discord channel weekly sync call on 25/07/2026. 

### AI Usage Disclosure:

We encourage contributors to use AI tools responsibly when creating Pull Requests. While AI can be a valuable aid, it is essential to ensure that your contributions meet the task requirements, build successfully, include relevant tests, and pass all linters. Submissions that do not meet these standards may be closed without warning to maintain the quality and integrity of the project. Please take the time to understand the changes you are proposing and their impact. AI slop is strongly discouraged and may lead to banning and blocking. Do not spam our repos with AI slop.

Check one of the checkboxes below:

- [ ] This PR does not contain AI-generated code at all.
- [x] This PR contains AI-generated code. I have read the [AI Usage Policy](https://github.com/AOSSIE-Org/Info/blob/main/AI-UsagePolicy.md) and this PR complies with this policy. I have tested the code locally and I am responsible for it.

I have used the following AI models and tools: Claude

## Checklist

- [x] My PR addresses a single issue, fixes a single bug or makes a single improvement.
- [x] My code follows the project's code style and conventions
- [x] If applicable, I have made corresponding changes or additions to the documentation
- [x] If applicable, I have made corresponding changes or additions to tests
- [x] My changes generate no new warnings or errors
- [x] I have joined the [Discord server](https://discord.gg/hjUhu33uAn) and I will share a link to this PR with the project maintainers there
- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Once I submit my PR, CodeRabbit AI will automatically review it and I will address CodeRabbit's comments.
- [x] I have filled this PR template completely and carefully, and I understand that my PR may be closed without review otherwise.

## Summary by CodeRabbit

* **Tests**
  * Expanded end-to-end verification for the models area: status reporting, hardware detection payloads, download/progress streaming via SSE, and setup flow.
  * Added strong checks for response shape and constraints (including task lifecycle states, tier validation, installation status, and numeric/allowed-value rules).
  * Improved reliability with isolated app mounting and automated reset of model download state between tests.
* **Chores**
  * Updated database table initialization SQL formatting without changing schema or behavior.

### CODE REVIEW

File: backend/tests/Bruno/PictoPy/Albums/Create Album.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        data: |-
+          {
+            "detail": [
+              {
+                "loc": [],
+                "msg": "",
+                "type": ""
+              }
+            ]
+          }
```
[the remark this answers is not in the document]
Comment (akshajtiwari) [reply backend/tests/Bruno/PictoPy/Albums/Create Album.yml]: tests will be added in a different PR , when all the test in the pytest module are converted for here and included for testing in BRUNO

**Comment by akshajtiwari** [comment]
Ready for merge @rohan-pandeyy
